bootstrap/src/sun/nio/ch/lincheck/Injections.java

```diff
@@ -176,6 +176,22 @@ public static void afterThreadFinish() {
         tracker.afterThreadFinish();
     }
 
+    /**
+     * Called from instrumented code when the thread finishes with an exception.
+     *
+     * @param exception the exception that was thrown in the current thread.
+     * @return true if the exception was handled by the tracker and should be suppressed, false otherwise.
+     */
+    public static boolean onThreadException(Throwable exception) {
+        Thread thread = Thread.currentThread();
+        // TestThread is handled separately
+        if (thread instanceof TestThread) return false;
```

Comment:
Should we handle this case separately?

Reply:
There are already several places where we handle `TestThread` differently. 
I've created an issue to fix this: #540.